src/services/room-service.js

```diff
@@ -96,20 +76,45 @@ export default class RoomService {
     return updatedRoom;
   }
 
-  async findOwnedRoomById({ roomId, ownerId }) {
+  async deleteRoom(roomId, user) {
     const room = await this.roomStore.findOne({ _id: roomId });
-    if (room?.owner !== ownerId) {
-      throw new NotFound(`A room with ID '${roomId}' owned by '${ownerId}' could not be found`);
+
+    if (!room) {
+      throw new NotFound();
+    }
+
+    if (room.owner !== user._id) {
+      throw new Forbidden();
+    }
+
+    await this.transactionRunner.run(async session => {
+      await this.lessonStore.deleteMany({ roomId }, { session });
+      await this.roomInvitationStore.deleteMany({ roomId }, { session });
+      await this.roomStore.deleteOne({ _id: roomId }, { session });
```

Comment:
For me it would be nicer if this filter mapping (`{ _id: roomId }`) would be done by the store, and the store would expose a `deleteRoomById({ roomId, session })` method.
Then one by one we could do the same for the other stores used in here too.

Reply:
Sounds good to me, I also thought about it, but was not sure how far we wnat to go with creating specialized functions, but I do like to have these `...ById` methods. I will add it.

Reply:
Done, I also created special store functions for most use cases (except `save`)

Reply:
Update: All generic functions are not used anymore and `RoomStore` and `RoomInvitationStore` do not inherit from `StoreBase` anymore.